### PR TITLE
feat(playreason): Play reason page

### PR DESCRIPTION
**Pull request by mitchfaulcon**
### Purpose

Added a new page between the charts dashboard and game itself for the user to enter the reason they are playing the game.

[HRT-49](https://softeng761team5.atlassian.net/browse/HRT-49)

### Approach

- Added radio buttons with an extra text box that appears when "Other" is selected
- Checks to make sure an option is selected. If "Other" is selected, a non-whitespace entry must be provided in the textbox
- The back button on the first dimension of the game routes back to the appropriate page (PlayReason if it's a new chart, charts dashboard page if a chart was clicked on, or the main dashboard if the user manually navigated to the game page)

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

N/A

#### Screenshot(s) of UI after this PR

image
image

### CODE REVIEW

**Review by michellemtan — DISMISSED** [review]
Looks good so far - just a few styling things! Could you please put the radio button styling into a css file for the PlayReason page for consistency with the rest of the other pages?

Also, I tried the page on my monitors and it looks good but could you add a bit more spacing between the reasons and the buttons? 

image

There's quite a bit of room on the card and the buttons are a little close to the reasons

**Review by gargi-bansal — CHANGES_REQUESTED** [review]
In terms of UI, maybe just keep the text area always visible but disable it initially. Only enable it when other is selected.
And I agree with Michelle. There should be a bit more spacing between the options and the buttons

File: src/pages/PlayReason.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect } from "react";
+import { withRouter, RouteComponentProps, useParams } from "react-router-dom";
+import { useCookies } from "react-cookie";
+import * as QueryString from "query-string";
+
+import { Button, Layout, Select, Typography, Radio } from "antd";
```
Comment (gargi-bansal) [inline src/pages/PlayReason.tsx]: unused import statement

File: src/pages/PlayReason.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+interface ParamTypes {
+  chartID: string;
+}
+
+const NONE_SELECTED = "A reason for playing the HeaRT Game must be selected";
+const NO_TEXT = "Please type in a reason for playing the HeaRT Game";
+
+const PlayReason: React.FC<RouteComponentProps> = (props) => {
+  const [cookies] = useCookies(["accessToken"]);
+  const { chartID } = useParams<ParamTypes>();
```
Comment (gargi-bansal) [inline src/pages/PlayReason.tsx]: unused variable? I don't think this or the ParamTypes interface is needed at all then?

File: src/pages/PlayReason.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  const onCancelClick = () => {
+    // Redirect back to previous course charts page
+    props.history.push(`/Course/${params.courseID}`);
+  };
+
+  return (
+    <div className="Course-Info">
+      <Header />
+
+      <Layout.Content className="Main-Container">
```
Comment (gargi-bansal) [inline src/pages/PlayReason.tsx]: I think you could honestly just use a div here?

**Review by gargi-bansal — CHANGES_REQUESTED** [review]
UI requests:
- Buttons together should be the same width as the text input
- Padding of 60px applied to whole form
- Consistent font size for the options and text area
- Update all CSS styling in reason of play to be consistent with what is already in master.
- Rename Form-Container in PlayReason.css to something else unique. It is identical to Form-Container in CreateCourse and thus changes the styling on that page.

**Review by gargi-bansal — APPROVED** [review]

**Review by lucycy98 — APPROVED** [review]
LGTM @mitchfaulcon!

**Pull request merged**